spdx-utils/src/main/kotlin/model/SpdxPackage.kt

```diff
@@ -0,0 +1,210 @@
+     * [SpdxConstants.NOASSERTION] must be used.
+     */
+    val licenseDeclared: String,
+
+    /**
+     * The license information found in the package as SPDX expression. To represent a not present value
+     * [SpdxConstants.NONE] or [SpdxConstants.NOASSERTION] must be used.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val licenseInfoFromFiles: List<String> = emptyList(),
+
+    /**
+     * The name of the package.
+     */
+    val name: String,
+
+    /**
+     * Identifies from where or whom the package originally came. The value must be a single line text in one of the
+     * following formats:
+     *
+     * 1. "Person: person name" or "Person: person name (email)"
+     * 2. "Organization: organization name" or "Organization: organization name (email)"
+     *
+     * TODO: Introduce a data type for above subjects.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val originator: String = "",
+
+    /**
+     * The actual file name of the package, or path of the directory being treated as a package.
+     */
+    @JsonProperty("packageFileName")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val packageFilename: String = "",
+
+    /**
+     * The [SpdxPackageVerificationCode] for the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val packageVerificationCode: SpdxPackageVerificationCode? = null,
+
+    /**
+     * Any relevant background information or additional comments about the origin of the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val sourceInfo: String = "",
+
+    /**
+     * A short description of the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val summary: String = "",
+
+    /**
+     * The distribution source for the package. The value must be a single line text in one of the
+     * following formats:
+     *
+     * 1. "Person: person name" or "Person: person name (email)"
+     * 2. "Organization: organization name" or "Organization: organization name (email)"
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val supplier: String = "",
+
+    /**
+     * The version of the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val versionInfo: String = ""
+) {
+    init {
+        require(spdxId.isNotBlank()) { "The SPDX-Id must not be blank." }
+
+        require(copyrightText.isNotBlank()) { "The copyright text must not be blank." }
+
+        require(downloadLocation.isNotBlank()) { "The download location must not be blank." }
+
+        require(name.isNotBlank()) { "The name must not be blank." }
+
+        require(homepage.isNotBlank()) { "The homepage must not be blank." }
+
+        // TODO: The check for [licenseInfoFromFiles] can be made more strict, but the SPDX specification is not exact
+        // enough yet to do this safely.
+        licenseInfoFromFiles.filterNot { it.isSpdxExpressionOrNotPresent() }.let {
+            require(it.isEmpty()) {
```

Comment:
Again leverage `all {}`.

Reply:
Doing so would either introduce redundant code for the error message or make the error message not mentioned the wrong elements. I prefer to keep it as is.